src/source-af-packet.c

```diff
@@ -1211,13 +1413,16 @@ TmEcode ReceiveAFPLoop(ThreadVars *tv, void *data, void *slot)
                 continue;
             }
         } else if (r > 0) {
-            if (ptv->flags & AFP_RING_MODE) {
-                r = AFPReadFromRing(ptv);
-            } else {
-                /* AFPRead will call TmThreadsSlotProcessPkt on read packets */
-                r = AFPRead(ptv);
-            }
+            r = AFPReadFunc(ptv);
             switch (r) {
+                case AFP_READ_OK:
+                    /* Trigger one dump of stats every second */
+                    TimeGet(&current_time);
```

Comment:
Do I read it correctly that with a block size of 4k and a frame size of 2k, AFPReadFromRingV3 will read 2 packets before returning? In that case any call to a time func may be expensive, esp if we can just keep the seconds from the packets we just processed in a local var.